chainer/function_hooks/cupy_memory_profile.py

```diff
@@ -0,0 +1,165 @@
+import sys
+
+from chainer import function
+
+from cupy.cuda import memory_hook
```

Comment:
It seems this line raises an error when CuPy is not imported. So, could you check the availability of CuPy and do some fallback if it is not found?